### PR TITLE
Add ability to delete Tool Servers

### PR DESCRIPTION
**Pull request by aiyakitori**
## What does this PR do?

Add new delete API for tool server.
Add delete button under Tool Server page.

image

## Checklists

- [x] Tests have been run locally and passed
- [x] New tests have been added to any work in /lib

## Summary by CodeRabbit

* **New Features**
  * Delete a Tool Server directly from its page with a confirmation dialog; successful deletion returns to the Tools list.
  * Backend supports deleting a Tool Server via the project/tool-server endpoint.

* **Improvements**
  * Clearer "Tool server not found" and "Project not found" error messages.
  * Tool Servers and Available Tools lists update immediately after deletions.
  * Better handling of diverse tool-server configurations (long names, Unicode, complex headers, query params, whitespace).

* **Tests**
  * Expanded coverage for create/update/retrieve/delete flows, multi-server scenarios, and edge cases.

### CODE REVIEW

File: app/web_ui/src/routes/(app)/settings/manage_tools/[project_id]/tool_servers/[tool_server_id]/+page.svelte [hunk 1]
```diff
@@ -322,3 +337,10 @@
     {/if}
   </AppPage>
 </div>
+
+<DeleteDialog
+  name="Tool Server"
+  bind:this={delete_dialog}
+  {delete_url}
```
Comment (scosman) [inline app/web_ui/src/routes/(app)/settings/manage_tools/[project_id]/tool_servers/[tool_server_id]/+page.svelte:341]: you make delete_url dynamic with `$:` but just attach it statically so the component won't get dyanmic updates.

You probably want:
```
bind:delete_url={`/api/projects/${project_id}/tool_servers/${tool_server_id}`}
```

File: app/web_ui/src/routes/(app)/settings/manage_tools/[project_id]/tool_servers/[tool_server_id]/+page.svelte [hunk 2]
```diff
@@ -8,13 +8,19 @@
...
   $: tool_server_id = $page.params.tool_server_id
 
   let tool_server: ExternalToolServerApiDescription | null = null
   let loading = true
   let error: KilnError | null = null
+  let delete_dialog: DeleteDialog | null = null
+  $: delete_url = `/api/projects/${project_id}/tool_servers/${tool_server_id}`
+  let after_delete = () => {
+    goto(`/settings/manage_tools/${project_id}`)
+  }
```
Comment (scosman) [inline app/web_ui/src/routes/(app)/settings/manage_tools/[project_id]/tool_servers/[tool_server_id]/+page.svelte]: These should be inline. Better to have the definitions where they are used for readability.

**Review by scosman — CHANGES_REQUESTED** [review]
small things

File: app/web_ui/src/routes/(app)/settings/manage_tools/[project_id]/tool_servers/[tool_server_id]/+page.svelte (hunk 1 above)
Comment (aiyakitori) [reply app/web_ui/src/routes/(app)/settings/manage_tools/[project_id]/tool_servers/[tool_server_id]/+page.svelte:341]: image
Svelte doesn't allow me to do this and without $: it doesn't get loaded properly since both required properties are bind dynamically. 

```
  $: project_id = $page.params.project_id
  $: tool_server_id = $page.params.tool_server_i
```

File: app/web_ui/src/routes/(app)/settings/manage_tools/[project_id]/tool_servers/[tool_server_id]/+page.svelte (hunk 2 above)
Comment (aiyakitori) [reply app/web_ui/src/routes/(app)/settings/manage_tools/[project_id]/tool_servers/[tool_server_id]/+page.svelte]: Done. 

We have 3 other places in the code that followed this pattern. 

```
<DeleteDialog
  name="Dataset Run"
  bind:this={delete_dialog}
  {delete_url}
  {after_delete}
/>
```

Comment (scosman) [reply app/web_ui/src/routes/(app)/settings/manage_tools/[project_id]/tool_servers/[tool_server_id]/+page.svelte]: heh. Well this is better.

Would be good to do delete_url too, but P3 (I was wrong about reactivity not working)

**Review by scosman — APPROVED** [review]
